### ISSUE TITLE
Bug: Issue with overriding the options for the `no-console` rule

### ISSUE THREAD

**Issue by srmagura** [eslint/eslint#17452]
### Environment

Node version: 18.15.0
npm version: 9.8.1
Local ESLint version: 8.46.0
Global ESLint version: N/A
Operating System: macOS 13.5

### What parser are you using?

Default (Espree)

### What did you do?

Minimal repro: https://github.com/srmagura/eslint-no-console-repro

<details>
<summary>Configuration</summary>

```js
//
// .eslintrc.base.cjs
//
module.exports = {
  rules: {
    "no-console": ["warn", { allow: ["warn", "error"] }],
  },
};

//
// .eslintrc.cjs
//
module.exports = {
  extends: ["./.eslintrc.base.cjs"],
  rules: {
    "no-console": "warn",

    // This configuration is considered invalid because the `allow` array is empty:

    // "no-console": ["warn", { allow: [] }],
  },
};
```
</details>

```js
console.log("this should be a warning");
console.warn("this should be a warning");
console.error("this should be a warning");
```

### What did you expect to happen?

All 3 console statements produce an ESLint warning.

### What actually happened?

Only the `console.log` statement produces an ESLint warning.

### Link to Minimal Reproducible Example

https://github.com/srmagura/eslint-no-console-repro

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

Sorry if this is a known issue which is fixed by the new config system!

**Comment by Rec0iL99** [comment]
Hi @srmagura, thanks for the issue. This is not a bug. You can override the inherited rule options by passing an empty object.

```js
"no-console": ["warn", {}],
```

**Resulting configuration from the `--print-config` CLI option:**

```bash
{
  "env": {},
  "globals": {},
  "parser": null,
  "parserOptions": {},
  "plugins": [],
  "rules": {
    "no-console": [
      "warn",
      {}
    ]
  },
  "settings": {},
  "ignorePatterns": []
}
```

**Result:**

```bash
npx eslint index.js                                                                                                                                                                                                        

/Users/joelmathew/WebProjects/eslint-misc/eslint-no-console-repro/index.js
  1:1  warning  Unexpected console statement  no-console
  2:1  warning  Unexpected console statement  no-console
  3:1  warning  Unexpected console statement  no-console

✖ 3 problems (0 errors, 3 warnings)
```

This works too:

```js
"no-console": ["warn", { allow: [""] }]
```
